config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -188,6 +190,14 @@ final ExpressionBasedFilterInvocationSecurityMetadataSource createMetadataSource
 			if (trustResolver != null) {
 				defaultHandler.setTrustResolver(trustResolver);
 			}
+			ApplicationContext context = http.getSharedObject(ApplicationContext.class);
+			if(context != null) {
```

Comment:
I'm wondering if we need to do this null check? The context is auto-wired into WebSecurityConfigurerAdapter and shared when getHttp() returns. If it's not set as a shared object then we would want a NPE to be thrown so we can detect the programming error. What do you think about removing the null check?